baremetal/metal3datatemplate_manager_test.go

```diff
@@ -444,6 +444,66 @@ var _ = Describe("Metal3DataTemplate manager", func() {
 		}),
 	)
 
+	type testCaseTemplateReference struct {
+		template1               *infrav1.Metal3DataTemplate
+		template2               *infrav1.Metal3DataTemplate
+		dataClaim               *infrav1.Metal3DataClaim
+		indexes                 map[int]string
+		expectError             bool
+		expectTemplateReference bool
+	}
+
+	DescribeTable("Test Template Reference",
+		func(tc testCaseTemplateReference) {
+			objects := []runtime.Object{}
+			objects = append(objects, tc.dataClaim)
+			c := fakeclient.NewFakeClientWithScheme(setupSchemeMm(), objects...)
+			templateMgr, err := NewDataTemplateManager(c, tc.template2,
+				klogr.New(),
+			)
+			Expect(err).NotTo(HaveOccurred())
+
+			_, err = templateMgr.createData(context.TODO(), tc.dataClaim, tc.indexes)
+			if tc.expectError {
+				Expect(err).To(HaveOccurred())
+			} else {
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			if tc.expectTemplateReference {
+				dataObjects := infrav1.Metal3DataList{}
+				opts := &client.ListOptions{}
+				err = c.List(context.TODO(), &dataObjects, opts)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(len(dataObjects.Items)).To(Equal(1))
+				Expect(dataObjects.Items[0].Spec.TemplateReference).To(Equal(tc.template1.Name))
+			}
+		},
+		Entry("TemplateReferenceExist", testCaseTemplateReference{
```

Comment:
could you please add more entries, to check the three cases : template reference matching, template reference and name matching (backwards compatibility), and no match